### PR TITLE
fix(hybrid-cloud): Updates sentry avatar query to use RPC service

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Adds a new `user_service` RPC method to retrieve user avatars by user ID. This new RPC call is used by the `sentry-avatar` module when querying and populating avatar information in emails.

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/user/impl.py [hunk 1]
```diff
@@ -252,6 +254,14 @@ def verify_user_emails(
             results[user_id] = RpcVerifyUserEmail(email=email, exists=exists)
         return results
 
+    def get_user_avatar(self, *, user_id: int) -> RpcAvatar | None:
+        possible_avatar = UserAvatar.objects.filter(user_id=user_id).first()
```
Comment (GabeVillalobos) [inline src/sentry/services/hybrid_cloud/user/impl.py:258]: Should this be a `get_from_cache` instead of a filter query?

Comment (markstory) [reply src/sentry/services/hybrid_cloud/user/impl.py:258]: You could use `get_from_cache`, but we would need to update the `UserAvatar` model as well. Currently it only supports cached reads on the `user` relation, not on `user_id`.

File: src/sentry/services/hybrid_cloud/user/serial.py [hunk 2]
```diff
@@ -79,12 +79,7 @@ def serialize_rpc_user(user: User) -> RpcUser:
     else:
         orm_avatar = user.avatar.first()
         if orm_avatar is not None:
-            avatar = RpcAvatar(
-                id=orm_avatar.id,
-                file_id=orm_avatar.file_id,
-                ident=orm_avatar.ident,
-                avatar_type=orm_avatar.get_avatar_type_display(),
-            )
+            avatar = serialize_user_avatar(avatar=orm_avatar)
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/user/serial.py:82]: If the avatar is serialized onto users, do we need a different RPC method? We could fetch the user with the avatar. :shrug:

**Review by markstory — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/user/serial.py (hunk 2 above)
Comment (GabeVillalobos) [reply src/sentry/services/hybrid_cloud/user/serial.py:82]: That's true and I did think about that, but figured it was worth cutting out the extra DB queries for the user and related entities. If we're not worried about the performance implication of this, I'm fine just using the get user RPC instead.

Comment (markstory) [reply src/sentry/services/hybrid_cloud/user/serial.py:82]: I think having a smaller dedicated RPC makes sense. User includes a bunch of data we're not going to user here.

**Pull request merged**